pkg/internal/testabilities/testservices/fixture_bhs.go

```diff
@@ -0,0 +1,113 @@
+package testservices
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/defs"
+	"github.com/go-softwarelab/common/pkg/to"
+	"github.com/jarcoal/httpmock"
+)
+
+type longestChainTipResponse struct {
+	Height        uint   `json:"height"`
+	Hash          string `json:"hash"`
+	Version       uint32 `json:"version"`
+	MerkleRoot    string `json:"merkleRoot"`
+	Timestamp     uint64 `json:"creationTimestamp"`
+	Bits          uint64 `json:"bits"`
+	Nonce         uint32 `json:"nonce"`
+	PreviousBlock string `json:"prevBlockHash"`
+}
+
+type LongestChainTipOptions func(*longestChainTipResponse)
+
+func WithLongestChainTipHeight(h uint) LongestChainTipOptions {
+	return func(l *longestChainTipResponse) {
+		l.Height = h
+	}
+}
+
+type BHSFixture interface {
+	IsUpAndRunning() BHSFixture
+	WillBeUnreachable() error
+	WillRespondWithInternalFailure()
+	WillRespondWithEmptyLongestTipBlockHeader()
+	OnLongestTipBlockHeaderResponseWith(opts ...LongestChainTipOptions)
+}
+
+type bhsFixture struct {
+	testing.TB
+	transport       *httpmock.MockTransport
+	longestChainTip *longestChainTipResponse
+}
+
+func (b bhsFixture) WillRespondWithEmptyLongestTipBlockHeader() {
```

Comment:
Struct bhsFixture has methods on both value and pointer receivers. Such usage is not recommended by the Go Documentation.

```suggestion
func (b *bhsFixture) WillRespondWithEmptyLongestTipBlockHeader() {
```

Reply:
good catch missed this :)